src/com/serotonin/mango/web/dwr/DataSourceListDwr.java

```diff
@@ -112,7 +114,7 @@ public int compare(Type o1, Type o2) {
 	}
 
 	public int deleteDataSource(int dataSourceId) {
-		Permissions.ensureDataSourcePermission(Common.getUser(), dataSourceId);
+		Permissions.ensureAdmin();
```

Comment:
Using "ensureAdmin()" we are limiting the usage of this "deleteDataSource" method only to administrators. Are we sure that we do not want to permit users with DataSource permission also to delete a DataSource?

Reply:
No user except admin can add / remove datasource.